lib/src/main/java/io/ably/lib/types/Capability.java

```diff
@@ -19,21 +19,19 @@
 public class Capability {
 
     /**
-     * Convenience method to canonicalise a JSON capability expression
+     * Convenience method to canonicalise a JSON capability expression.
      *
      * @param capability a capability string, which is the JSON text for the capability
      * @return a capability string which has been canonicalised
      * @throws AblyException if there is an error processing the given string
      * (if for example it is not valid JSON)
      */
-    public static final String c14n(String capability) throws AblyException {
-        if(capability == null || capability.isEmpty()) return "";
+    public static String c14n(String capability) throws AblyException {
+        if (capability == null || capability.isEmpty()) return capability;
```

Comment:
I feel it should return null when the capability is empty too, doesn't make sense to send empty capability to the server 🤔

Reply:
@sacOO7, can you reference something in the spec or perhaps another client library implementation that does exhibit the behaviour you describe?

Reply:
Okay, let me check.

Reply:
https://github.com/ably/ably-go/blob/1df14a2afaad30eaac619fb36a88bafef654f0b8/ably/token.go#L22
says it to be `omitempty`
- Means it will not include the key if the capability is `empty` or `null`  (https://www.sohamkamani.com/golang/omitempty/)

Reply:
Updated the code to return null for empty strings

Reply:
Great, thanks both. 😁